File: ios/NuvioTV/Sources/Models/CatalogModels.swift
//
//  CatalogModels.swift
//  NuvioTV
//
//  Created by Claude Code
//  Swift data models for catalog browsing
//

import Foundation

// MARK: - Catalog Models

/// Catalog collection with items
struct Catalog: Identifiable, Codable {
    let id: String
    let name: String
    let description: String
    let itemIds: [String]
}

/// Content metadata
struct Meta: Identifiable, Codable {
    let id: String
    let name: String
    let description: String?
    let posterUrl: String?
    let backgroundUrl: String?
    let logoUrl: String?
    let imdbId: String?
    let tmdbId: Int?
    let type: String
    let year: Int?
    let genres: [String]?
    let rating: Double?
    let releaseInfo: String?
    let runtime: String?
    let cast: [String]?
    let director: [String]?
    let writer: [String]?
    let certification: String?
    let country: String?
    let released: String?
}

/// Video stream information
struct Stream: Identifiable, Codable {
    var id: String { url ?? UUID().uuidString }
    let url: String?
    let name: String?
    let description: String?
    let addonName: String?
}

/// Paginated catalog page
struct CatalogPage {
    let items: [Meta]
    let hasMore: Bool
    let page: Int
}

// MARK: - Filter & Sort Models

/// Filter state for catalog browsing
struct FilterState: Equatable {
    var contentType: String = "movie"
    var genre: String? = nil
    var year: Int? = nil
    var sort: SortOption = .trending
}

/// Sort options for catalog
enum SortOption: String, CaseIterable {
    case trending = "top"
    case popular = "popular"
    case newest = "newest"
    case rating = "rating"

    var displayName: String {
        switch self {
        case .trending: return "Trending"
        case .popular: return "Popular"
        case .newest: return "Newest"
        case .rating: return "Top Rated"
        }
    }

    var catalogId: String {
        return self.rawValue
    }
}

// MARK: - UI State

/// UI state for catalog browse screen
struct CatalogBrowseUiState {
    var isLoading: Bool = false
    var items: [Meta] = []
    var currentPage: Int = 1
    var hasMore: Bool = true
    var filterState: FilterState = FilterState()
    var availableGenres: [String] = []
    var error: String? = nil
    var isLoadingMore: Bool = false
}

/// UI state for details screen
struct DetailsUiState {
    var isLoading: Bool = true
    var meta: Meta? = nil
    var streams: [Stream] = []
    var error: String? = nil
    var isInWatchlist: Bool = false
    var userRating: Int? = nil
}

File: android/app/src/main/java/com/nuvio/app/tv/ui/catalog/CatalogModels.kt
package com.nuvio.app.tv.ui.catalog

import com.nuvio.app.tv.data.repository.Meta

/**
 * Filter state for catalog browsing
 */
data class FilterState(
    val contentType: String = "movie",
    val genre: String? = null,
    val year: Int? = null,
    val sort: SortOption = SortOption.TRENDING
)

/**
 * Sort options for catalog
 */
enum class SortOption(val displayName: String, val catalogId: String) {
    TRENDING("Trending", "top"),
    POPULAR("Popular", "popular"),
    NEWEST("Newest", "newest"),
    RATING("Top Rated", "rating")
}

/**
 * UI state for catalog browse screen
 */
data class CatalogBrowseUiState(
    val isLoading: Boolean = false,
    val items: List<Meta> = emptyList(),
    val currentPage: Int = 1,
    val hasMore: Boolean = true,
    val filterState: FilterState = FilterState(),
    val availableGenres: List<String> = emptyList(),
    val error: String? = null,
    val isLoadingMore: Boolean = false
)
